From f76b9267392f77716d205394108e1d9fe156c416 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Sat, 5 Sep 2026 18:42:19 +0200
Subject: [PATCH] parpack: optional installation

In MSYS2, the parpack library is split into a separate package.

Support using the CMake config files in the case where the parpack package is
not installed.
---
 CMakeLists.txt                 | 7 ++++++-
 cmake/arpackng-config.cmake.in | 4 ++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d15ff0e..c46b63f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -860,7 +860,7 @@ install(FILES "${PROJECT_BINARY_DIR}/SRC/arpack${LIBSUFFIX}${ITF64SUFFIX}.pc"
 
 if (MPI)
   install(TARGETS parpack
-      EXPORT arpackngTargets
+      EXPORT parpackngTargets
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
@@ -905,6 +905,11 @@ endif()
 install(EXPORT arpackngTargets
   DESTINATION "${ARPACK_INSTALL_CMAKEDIR}"
 )
+if (MPI)
+  install(EXPORT parpackngTargets
+    DESTINATION "${ARPACK_INSTALL_CMAKEDIR}"
+)
+endif()
 # Provide find_package for arpack-ng to users.
 configure_file(cmake/arpackng-config.cmake.in "${PROJECT_BINARY_DIR}/arpackng-config.cmake" @ONLY)
 configure_file(cmake/arpackng-config-version.cmake.in "${PROJECT_BINARY_DIR}/arpackng-config-version.cmake" @ONLY)
diff --git a/cmake/arpackng-config.cmake.in b/cmake/arpackng-config.cmake.in
index bafa3fe..72e5458 100644
--- a/cmake/arpackng-config.cmake.in
+++ b/cmake/arpackng-config.cmake.in
@@ -34,6 +34,10 @@ endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/arpackngTargets.cmake")
 
+if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/parpackngTargets.cmake")
+  include("${CMAKE_CURRENT_LIST_DIR}/parpackngTargets.cmake")
+endif()
+
 add_library(ARPACK::ARPACK ALIAS arpack)
 if (TARGET parpack)
 	add_library(PARPACK::PARPACK ALIAS parpack)
-- 
2.51.0.windows.2

